src/bin/commands/demux.rs

```diff
@@ -243,6 +272,8 @@ struct ReadSetIterator {
     read_structure: ReadStructure,
     /// The file containing FASTQ records.
     source: FastqReader<Box<dyn BufRead + Send>>,
+    /// Valid reasons for skipping reads, otherwise panic!
+    skip_reasons: HashSet<SkipReason>,
```

Comment:
I would think that for a set of reasons that is likely very short (usually 0-1) that a Vec[SkipReason] would be faster.